src/compiler/checker.ts

```diff
@@ -4656,6 +4662,10 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 return getMergedSymbol(sourceFile.symbol);
             }
             if (errorNode && moduleNotFoundError) {
+                if (isSideEffectImport) {
+                    // This a side-effect import; ignore the fact that it's not a module.
+                    return undefined;
+                }

```

Comment:
```suggestion
            if (errorNode && moduleNotFoundError && !isSideEffectImport) {
```

Reply:
Did this (more or less); fixed a case where the public API would misbehave with this.